### PR TITLE
API fixes: escape characters on url and query strings

### PR DESCRIPTION
**Pull request by elorzafe**
*Issue #, if available:*
Fixes: #361 
Fixes: #127
*Description of changes:*
Escape characters on url and query strings fix

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: packages/aws-amplify/package.json [hunk 1]
```diff
@@ -1,13 +1,13 @@
 {
   "name": "aws-amplify",
-  "version": "0.3.3",
+  "version": "0.3.4",
```
Comment (manueliglesias) [inline packages/aws-amplify/package.json]: Let's revert this change until the actual publish happens

File: packages/aws-amplify/src/API/RestClient.ts [hunk 2]
```diff
@@ -111,7 +112,7 @@ export class RestClient {
         }
 
         return Auth.currentCredentials()
-            .then(credentials => this._signed(params, credentials, isAllResponse));
+            .then(credentials => this._signed(Object.assign({},params,extraParams), credentials, isAllResponse));
```
Comment (manueliglesias) [inline packages/aws-amplify/src/API/RestClient.ts]: To be consistent with lines below, lets use the spread operator instead of Object.assign

`Object.assign({},params,extraParams)` => `{...params, ...extraParams}`

File: packages/aws-amplify/src/Common/Polyfills/Polyfills.ts [hunk 3]
```diff
@@ -27,3 +27,5 @@ if (!(<any>global).window) {
 if (!(<any>global).localStorage) {
     (<any>global).localStorage = InMemoryCache;
 }
+
+(<any>Object).entries = Object.entries || (x => Object.keys(x).map(k => [k, x[k]]));
```
Comment (manueliglesias) [inline packages/aws-amplify/src/Common/Polyfills/Polyfills.ts]: If this is not needed yet, lets remove it

File: packages/aws-amplify/tsconfig.json [hunk 4]
```diff
@@ -8,7 +8,8 @@
             "es5",
             "es2015",
             "dom",
-            "esnext.asynciterable"
+            "esnext.asynciterable",
+            "es2017.object"
```
Comment (manueliglesias) [inline packages/aws-amplify/tsconfig.json]: Was this added to get Object.entries ? If so, let's remove it too since we didn't use

File: packages/aws-amplify/src/API/RestClient.ts [hunk 5]
```diff
@@ -111,7 +112,7 @@ export class RestClient {
         }
 
         return Auth.currentCredentials()
-            .then(credentials => this._signed(params, credentials, isAllResponse));
+            .then(credentials => this._signed({...params, ...extraParams}, credentials, isAllResponse));
```
Comment (j1-uli-holtel) [inline packages/aws-amplify/src/API/RestClient.ts]: The same or similar changes need to be applied  for the unsigned request (when the `Authorization` is provided).

Comment (elorzafe) [reply packages/aws-amplify/src/API/RestClient.ts]: Nice catch! I will send the pr for that fix before publishing. Thanks!
